src/libraries/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -127,88 +173,23 @@ private static unsafe bool SystemSupportsPreciseSystemTime()
                 // when we better understand what's causing the drift and the issue is no longer
                 // a problem or can be better worked around on all targeted OSes.
 
-                long systemTimeResult;
-                Interop.Kernel32.GetSystemTimeAsFileTime(&systemTimeResult);
-
-                long preciseSystemTimeResult;
-                Interop.Kernel32.GetSystemTimePreciseAsFileTime(&preciseSystemTimeResult);
-
-                return Math.Abs(preciseSystemTimeResult - systemTimeResult) <= 100 * TicksPerMillisecond;
-            }
-
-            return false;
-        }
-
-        private static unsafe bool ValidateSystemTime(Interop.Kernel32.SYSTEMTIME* time, bool localTime)
-        {
-            if (localTime)
-            {
-                Interop.Kernel32.SYSTEMTIME st;
-                return Interop.Kernel32.TzSpecificLocalTimeToSystemTime(IntPtr.Zero, time, &st) != Interop.BOOL.FALSE;
-            }
-            else
-            {
-                long timestamp;
-                return Interop.Kernel32.SystemTimeToFileTime(time, &timestamp) != Interop.BOOL.FALSE;
-            }
-        }
-
-        private static unsafe bool FileTimeToSystemTime(long fileTime, FullSystemTime* time)
-        {
-            if (Interop.Kernel32.FileTimeToSystemTime(&fileTime, &time->systemTime) != Interop.BOOL.FALSE)
-            {
-                // to keep the time precision
-                time->hundredNanoSecond = fileTime % TicksPerMillisecond;
-                if (time->systemTime.Second > 59)
-                {
-                    // we have a leap second, force it to last second in the minute as DateTime doesn't account for leap seconds in its calculation.
-                    // we use the maxvalue from the milliseconds and the 100-nano seconds to avoid reporting two out of order 59 seconds
-                    time->systemTime.Second = 59;
-                    time->systemTime.Milliseconds = 999;
-                    time->hundredNanoSecond = 9999;
-                }
-                return true;
-            }
-            return false;
-        }
-
-        private static unsafe void GetSystemTimeWithLeapSecondsHandling(FullSystemTime* time)
-        {
-            if (!FileTimeToSystemTime(GetSystemTimeAsFileTime(), time))
-            {
-                Interop.Kernel32.GetSystemTime(&time->systemTime);
-                time->hundredNanoSecond = 0;
-                if (time->systemTime.Second > 59)
+                // Retry this check several times to reduce chance of false negatives due to thread being rescheduled
+                // at wrong time.
+                for (int i = 0; i < 10; i++)
```

Comment:
This retry loop wasn't present previously, was it?

Reply:
Yes, I am adding it proactively to make this check more robust. I think we are incorrectly falling back to imprecise UtcNow like one in million runtime startups without this loop. The loop will exit in first iteration on well-behaving systems.